// react-cssfx-loading/dist/react-cssfx-loading.esm.js
import { css, keyframes } from 'goober';
import React from 'react';

function _objectWithoutPropertiesLoose(source, excluded) {
  if (source == null) return {};
  var target = {};
  var sourceKeys = Object.keys(source);
  var key, i;

  for (i = 0; i < sourceKeys.length; i++) {
    key = sourceKeys[i];
    if (excluded.indexOf(key) >= 0) continue;
    target[key] = source[key];
  }

  return target;
}

function _taggedTemplateLiteralLoose(strings, raw) {
  if (!raw) {
    raw = strings.slice(0);
  }

  strings.raw = raw;
  return strings;
}

var _excluded = ["className", "color", "width", "height", "style", "duration"];

var _templateObject, _templateObject2;
var grow = /*#__PURE__*/keyframes(_templateObject || (_templateObject = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n  0%, 100% {\n    transform: scaleY(1);\n  }\n\n  50% {\n    transform: scaleY(2);\n  }\n"])));

var BarWave = function BarWave(_ref) {
  var _ref$className = _ref.className,
      className = _ref$className === void 0 ? "" : _ref$className,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? "#0d6efd" : _ref$color,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? "2rem" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "1rem" : _ref$height,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? "1s" : _ref$duration,
      others = _objectWithoutPropertiesLoose(_ref, _excluded);

  var resolvedWidth = typeof width === "number" ? width + "px" : width;
  var resolvedHeight = typeof height === "number" ? height + "px" : height;
  return React.createElement("div", Object.assign({}, others, {
    style: style,
    className: css(_templateObject2 || (_templateObject2 = _taggedTemplateLiteralLoose(["\n          display: flex;\n          flex-flow: row nowrap;\n          align-items: center;\n          justify-content: space-between;\n          width: ", ";\n\n          & span {\n            width: calc(", " / 4 * 3 / 4);\n            height: ", ";\n            background-color: ", ";\n          }\n\n          & span:nth-of-type(1) {\n            animation: ", " ", " -0.45s ease-in-out infinite;\n          }\n\n          & span:nth-of-type(2) {\n            animation: ", " ", " -0.3s ease-in-out infinite;\n          }\n\n          & span:nth-of-type(3) {\n            animation: ", " ", " -0.15s ease-in-out infinite;\n          }\n\n          & span:nth-of-type(4) {\n            animation: ", " ", " ease-in-out infinite;\n          }\n        "])), resolvedWidth, resolvedWidth, resolvedHeight, color, grow, duration, grow, duration, grow, duration, grow, duration) + (" " + className)
  }), React.createElement("span", null), React.createElement("span", null), React.createElement("span", null), React.createElement("span", null));
};

var _excluded$1 = ["className", "color", "width", "height", "style", "duration"];

var _templateObject$1, _templateObject2$1, _templateObject3;
var leftSwing = /*#__PURE__*/keyframes(_templateObject$1 || (_templateObject$1 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n  50%,\n  100% {\n    transform: translateX(0);\n  }\n"])));
var rightSwing = /*#__PURE__*/keyframes(_templateObject2$1 || (_templateObject2$1 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n  50% {\n    transform: translateX(0);\n  }\n  100% {\n    transform: translateX(100%);\n  }\n"])));

var BouncingBalls = function BouncingBalls(_ref) {
  var _ref$className = _ref.className,
      className = _ref$className === void 0 ? "" : _ref$className,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? "#0d6efd" : _ref$color,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? "3rem" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "1rem" : _ref$height,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? "0.5s" : _ref$duration,
      others = _objectWithoutPropertiesLoose(_ref, _excluded$1);

  var resolvedWidth = typeof width === "number" ? width + "px" : width;
  var resolvedHeight = typeof height === "number" ? height + "px" : height;
  return React.createElement("div", Object.assign({}, others, {
    style: style,
    className: css(_templateObject3 || (_templateObject3 = _taggedTemplateLiteralLoose(["\n          display: inline-flex;\n          flex-flow: row nowrap;\n          align-items: center;\n          justify-content: space-between;\n          width: ", ";\n\n          & div {\n            width: calc(", " / 3);\n            height: ", ";\n            border-radius: 50%;\n            background-color: ", ";\n          }\n\n          & div:nth-of-type(1) {\n            transform: translateX(-100%);\n            animation: ", " ", " ease-in alternate infinite;\n          }\n\n          & div:nth-of-type(3) {\n            transform: translateX(0);\n            animation: ", " ", " ease-out alternate infinite;\n          }\n        "])), resolvedWidth, resolvedWidth, resolvedHeight, color, leftSwing, duration, rightSwing, duration) + (" " + className)
  }), React.createElement("div", null), React.createElement("div", null), React.createElement("div", null));
};

var _excluded$2 = ["className", "color", "width", "height", "style", "duration"];

var _templateObject$2, _templateObject2$2, _templateObject3$1, _templateObject4;
var rotate = /*#__PURE__*/keyframes(_templateObject$2 || (_templateObject$2 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n  100% {\n    transform: rotate(360deg);\n  }\n"])));
var dash = /*#__PURE__*/keyframes(_templateObject2$2 || (_templateObject2$2 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n  0% {\n    stroke-dasharray: 1, 200;\n    stroke-dashoffset: 0;\n  }\n  50% {\n    stroke-dasharray: 90, 200;\n    stroke-dashoffset: -35px;\n  }\n  100% {\n    stroke-dashoffset: -125px;\n  }\n"])));

var CircularProgress = function CircularProgress(_ref) {
  var _ref$className = _ref.className,
      className = _ref$className === void 0 ? "" : _ref$className,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? "#0d6efd" : _ref$color,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? "3rem" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "3rem" : _ref$height,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? "2s" : _ref$duration,
      others = _objectWithoutPropertiesLoose(_ref, _excluded$2);

  var resolvedWidth = typeof width === "number" ? width + "px" : width;
  var resolvedHeight = typeof height === "number" ? height + "px" : height;
  return React.createElement("svg", Object.assign({}, others, {
    crossOrigin: "anonymous",
    viewBox: "25 25 50 50",
    style: style,
    className: css(_templateObject3$1 || (_templateObject3$1 = _taggedTemplateLiteralLoose(["\n          width: ", ";\n          height: ", ";\n          transform-origin: center;\n          animation: ", " ", " linear infinite;\n        "])), resolvedWidth, resolvedHeight, rotate, duration) + (" " + className)
  }), React.createElement("circle", {
    className: css(_templateObject4 || (_templateObject4 = _taggedTemplateLiteralLoose(["\n          fill: none;\n          stroke: ", ";\n          stroke-width: 3;\n          stroke-dasharray: 1, 200;\n          stroke-dashoffset: 0;\n          stroke-linecap: round;\n          animation: ", " calc(", " * 3 / 4) ease-in-out infinite;\n        "])), color, dash, duration),
    cx: "50",
    cy: "50",
    r: "20"
  }));
};

var _excluded$3 = ["className", "color", "width", "height", "style", "duration"];

var _templateObject$3, _templateObject2$3, _templateObject3$2;
var spin = /*#__PURE__*/keyframes(_templateObject$3 || (_templateObject$3 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n  to {\n    transform: rotateY(540deg);\n  }\n"])));

var Coin = function Coin(_ref) {
  var _ref$className = _ref.className,
      className = _ref$className === void 0 ? "" : _ref$className,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? "#0d6efd" : _ref$color,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? "2rem" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "2rem" : _ref$height,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? "1.2s" : _ref$duration,
      others = _objectWithoutPropertiesLoose(_ref, _excluded$3);

  var resolvedWidth = typeof width === "number" ? width + "px" : width;
  var resolvedHeight = typeof height === "number" ? height + "px" : height;
  return React.createElement("div", Object.assign({}, others, {
    style: style,
    className: css(_templateObject2$3 || (_templateObject2$3 = _taggedTemplateLiteralLoose(["\n          perspective: 9999px;\n        "]))) + (" " + className)
  }), React.createElement("div", {
    className: css(_templateObject3$2 || (_templateObject3$2 = _taggedTemplateLiteralLoose(["\n          width: ", ";\n          height: ", ";\n          border-radius: 50%;\n          border: calc(", " / 8) solid ", ";\n          animation: ", " ", " ease-in-out infinite;\n        "])), resolvedWidth, resolvedHeight, resolvedWidth, color, spin, duration)
  }));
};

var _excluded$4 = ["className", "color", "width", "height", "style", "duration"];

var _templateObject$4, _templateObject2$4;
var fade = /*#__PURE__*/keyframes(_templateObject$4 || (_templateObject$4 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n  from {\n    opacity: 1;\n  }\n  to {\n    opacity: 0;\n  }\n"])));

var FadingBalls = function FadingBalls(_ref) {
  var _ref$className = _ref.className,
      className = _ref$className === void 0 ? "" : _ref$className,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? "#0d6efd" : _ref$color,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? "4rem" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "1rem" : _ref$height,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? "0.8s" : _ref$duration,
      others = _objectWithoutPropertiesLoose(_ref, _excluded$4);

  var resolvedWidth = typeof width === "number" ? width + "px" : width;
  var resolvedHeight = typeof height === "number" ? height + "px" : height;
  return React.createElement("div", Object.assign({}, others, {
    style: style,
    className: css(_templateObject2$4 || (_templateObject2$4 = _taggedTemplateLiteralLoose(["\n          width: ", ";\n          display: flex;\n          flex-flow: row nowrap;\n          align-items: center;\n          justify-content: space-between;\n\n          & div {\n            width: calc(", " / 4);\n            height: ", ";\n            border-radius: 50%;\n            background-color: ", ";\n            animation: ", " ", " ease-in-out alternate infinite;\n          }\n\n          & div:nth-of-type(1) {\n            animation-delay: calc(", " * -0.5);\n          }\n\n          & div:nth-of-type(2) {\n            animation-delay: calc(", " * -0.25);\n          }\n        "])), resolvedWidth, resolvedWidth, resolvedHeight, color, fade, duration, duration, duration) + (" " + className)
  }), React.createElement("div", null), React.createElement("div", null), React.createElement("div", null));
};

var _excluded$5 = ["className", "color", "width", "height", "style", "duration"];

var _templateObject$5, _templateObject2$5;
var fade$1 = /*#__PURE__*/keyframes(_templateObject$5 || (_templateObject$5 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n  to {\n    opacity: 0.2;\n  }\n"])));

var FadingDots = function FadingDots(_ref) {
  var _ref$className = _ref.className,
      className = _ref$className === void 0 ? "" : _ref$className,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? "#0d6efd" : _ref$color,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? "2rem" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "2rem" : _ref$height,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? "0.6s" : _ref$duration,
      others = _objectWithoutPropertiesLoose(_ref, _excluded$5);

  var resolvedWidth = typeof width === "number" ? width + "px" : width;
  var resolvedHeight = typeof height === "number" ? height + "px" : height;
  return React.createElement("div", Object.assign({}, others, {
    style: style,
    className: css(_templateObject2$5 || (_templateObject2$5 = _taggedTemplateLiteralLoose(["\n          width: ", ";\n          height: ", ";\n          display: grid;\n          grid-template-rows: repeat(3, 1fr);\n          grid-template-columns: repeat(3, 1fr);\n          justify-items: center;\n          align-items: center;\n\n          & > div {\n            width: calc(", " / 6);\n            height: calc(", " / 6);\n            background-color: ", ";\n            border-radius: 50%;\n            animation: ", " ", " alternate ease-in-out infinite;\n          }\n\n          & > div:nth-of-type(2),\n          & > div:nth-of-type(4) {\n            animation-delay: calc(", " / 6);\n          }\n\n          & > div:nth-of-type(3),\n          & > div:nth-of-type(5),\n          & > div:nth-of-type(7) {\n            animation-delay: calc(", " / 3);\n          }\n\n          & > div:nth-of-type(6),\n          & > div:nth-of-type(8) {\n            animation-delay: calc(", " / 2);\n          }\n\n          & > div:nth-of-type(9) {\n            animation-delay: calc(", " * 2 / 3);\n          }\n        "])), resolvedWidth, resolvedHeight, resolvedWidth, resolvedHeight, color, fade$1, duration, duration, duration, duration, duration) + (" " + className)
  }), React.createElement("div", null), React.createElement("div", null), React.createElement("div", null), React.createElement("div", null), React.createElement("div", null), React.createElement("div", null), React.createElement("div", null), React.createElement("div", null), React.createElement("div", null));
};

var _excluded$6 = ["className", "color", "width", "height", "style", "duration"];

var _templateObject$6, _templateObject2$6, _templateObject3$3;
var spin$1 = /*#__PURE__*/keyframes(_templateObject$6 || (_templateObject$6 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n  50%,\n  100% {\n    transform: rotate(360deg);\n  }\n"])));
var fill = /*#__PURE__*/keyframes(_templateObject2$6 || (_templateObject2$6 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n  25%,\n  50% {\n    transform: scaleY(0);\n  }\n  100% {\n    transform: scaleY(1);\n  }\n"])));

var FillingBottle = function FillingBottle(_ref) {
  var _ref$className = _ref.className,
      className = _ref$className === void 0 ? "" : _ref$className,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? "#0d6efd" : _ref$color,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? "2rem" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "2rem" : _ref$height,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? "1.4s" : _ref$duration,
      others = _objectWithoutPropertiesLoose(_ref, _excluded$6);

  var resolvedWidth = typeof width === "number" ? width + "px" : width;
  var resolvedHeight = typeof height === "number" ? height + "px" : height;
  return React.createElement("div", Object.assign({}, others, {
    style: style,
    className: css(_templateObject3$3 || (_templateObject3$3 = _taggedTemplateLiteralLoose(["\n          position: relative;\n          width: ", ";\n          height: ", ";\n          border: 3px solid ", ";\n          overflow: hidden;\n          animation: ", " ", " ease infinite;\n\n          &::before {\n            content: \"\";\n            position: absolute;\n            top: -3px;\n            left: -3px;\n            width: calc(100% + 3px);\n            height: calc(100% + 3px);\n            background-color: ", ";\n            opacity: 0.75;\n            transform-origin: center bottom;\n            transform: scaleY(1);\n            animation: ", " ", " linear infinite;\n          }\n        "])), resolvedWidth, resolvedHeight, color, spin$1, duration, color, fill, duration) + (" " + className)
  }));
};

var _excluded$7 = ["className", "color", "width", "height", "style", "duration"];

var _templateObject$7, _templateObject2$7, _templateObject3$4;
var flip = /*#__PURE__*/keyframes(_templateObject$7 || (_templateObject$7 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n  50% {\n    transform: rotateY(180deg);\n  }\n  100% {\n    transform: rotateY(180deg) rotateX(180deg);\n  }\n"])));

var FlippingSquare = function FlippingSquare(_ref) {
  var _ref$className = _ref.className,
      className = _ref$className === void 0 ? "" : _ref$className,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? "#0d6efd" : _ref$color,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? "2rem" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "2rem" : _ref$height,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? "1s" : _ref$duration,
      others = _objectWithoutPropertiesLoose(_ref, _excluded$7);

  var resolvedWidth = typeof width === "number" ? width + "px" : width;
  var resolvedHeight = typeof height === "number" ? height + "px" : height;
  return React.createElement("div", Object.assign({}, others, {
    style: style,
    className: css(_templateObject2$7 || (_templateObject2$7 = _taggedTemplateLiteralLoose(["\n          perspective: 9999px;\n        "]))) + (" " + className)
  }), React.createElement("div", {
    className: css(_templateObject3$4 || (_templateObject3$4 = _taggedTemplateLiteralLoose(["\n          width: ", ";\n          height: ", ";\n          background-color: ", ";\n          transform: rotate(0);\n          animation: ", " ", " infinite;\n        "])), resolvedWidth, resolvedHeight, color, flip, duration)
  }));
};

var _excluded$8 = ["className", "color", "width", "height", "style", "duration"];

var _templateObject$8, _templateObject2$8, _templateObject3$5, _templateObject4$1, _templateObject5;
var spin$2 = /*#__PURE__*/keyframes(_templateObject$8 || (_templateObject$8 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n  to {\n    transform: translate(-50%, -50%) rotate(360deg);\n  }\n"])));

var Hypnosis = function Hypnosis(_ref) {
  var _ref$className = _ref.className,
      className = _ref$className === void 0 ? "" : _ref$className,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? "#0d6efd" : _ref$color,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? "2rem" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "2rem" : _ref$height,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? "1.2s" : _ref$duration,
      others = _objectWithoutPropertiesLoose(_ref, _excluded$8);

  var resolvedWidth = typeof width === "number" ? width + "px" : width;
  var resolvedHeight = typeof height === "number" ? height + "px" : height;
  return React.createElement("div", Object.assign({}, others, {
    style: style,
    className: css(_templateObject2$8 || (_templateObject2$8 = _taggedTemplateLiteralLoose(["\n          position: relative;\n          width: ", ";\n          height: ", ";\n\n          & div {\n            border: 3px solid transparent;\n            border-top-color: ", ";\n            border-right-color: ", ";\n            border-radius: 50%;\n            position: absolute;\n            top: 50%;\n            left: 50%;\n            transform: translate(-50%, -50%) rotate(0);\n          }\n        "])), resolvedWidth, resolvedHeight, color, color) + (" " + className)
  }), React.createElement("div", {
    className: css(_templateObject3$5 || (_templateObject3$5 = _taggedTemplateLiteralLoose(["\n          width: ", ";\n          height: ", ";\n          animation: ", " ", " linear infinite;\n        "])), resolvedWidth, resolvedHeight, spin$2, duration)
  }), React.createElement("div", {
    className: css(_templateObject4$1 || (_templateObject4$1 = _taggedTemplateLiteralLoose(["\n          width: calc(", " * 0.6);\n          height: calc(", " * 0.6);\n          animation: ", " calc(", " * 7 / 8) linear reverse infinite;\n        "])), resolvedWidth, resolvedHeight, spin$2, duration)
  }), React.createElement("div", {
    className: css(_templateObject5 || (_templateObject5 = _taggedTemplateLiteralLoose(["\n          width: calc(", " * 0.8 / 3.5);\n          height: calc(", " * 0.8 / 3.5);\n          animation: ", " calc(", " * 3 / 4) linear infinite;\n        "])), resolvedWidth, resolvedHeight, spin$2, duration)
  }));
};

var _excluded$9 = ["className", "color", "width", "height", "style", "duration"];

var _templateObject$9, _templateObject2$9;
var wave = /*#__PURE__*/keyframes(_templateObject$9 || (_templateObject$9 = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n  from {\n    transform: translateY(-60%);\n  }\n  to {\n    transform: translateY(60%);\n  }\n"])));

var Messaging = function Messaging(_ref) {
  var _ref$className = _ref.className,
      className = _ref$className === void 0 ? "" : _ref$className,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? "#0d6efd" : _ref$color,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? "1rem" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "1rem" : _ref$height,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? "0.4s" : _ref$duration,
      others = _objectWithoutPropertiesLoose(_ref, _excluded$9);

  var resolvedWidth = typeof width === "number" ? width + "px" : width;
  var resolvedHeight = typeof height === "number" ? height + "px" : height;
  return React.createElement("div", Object.assign({}, others, {
    style: style,
    className: css(_templateObject2$9 || (_templateObject2$9 = _taggedTemplateLiteralLoose(["\n          display: flex;\n          flex-flow: row nowrap;\n          align-items: center;\n          gap: calc(", " * 0.5);\n\n          & div {\n            width: ", ";\n            height: ", ";\n            border-radius: 50%;\n            background-color: ", ";\n            transform: translateY(-100%);\n            animation: ", " ", " ease-in-out alternate infinite;\n          }\n\n          & div:nth-of-type(1) {\n            animation-delay: calc(", " * -0.5);\n          }\n\n          & div:nth-of-type(2) {\n            animation-delay: calc(", " * -0.25);\n          }\n        "])), resolvedWidth, resolvedWidth, resolvedHeight, color, wave, duration, duration, duration) + (" " + className)
  }), React.createElement("div", null), React.createElement("div", null), React.createElement("div", null));
};

var _excluded$a = ["className", "color", "width", "height", "style", "duration"];

var _templateObject$a, _templateObject2$a, _templateObject3$6, _templateObject4$2;
var spin$3 = /*#__PURE__*/keyframes(_templateObject$a || (_templateObject$a = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n  to {\n    transform: rotate(360deg);\n  }\n"])));

var Ring = function Ring(_ref) {
  var _ref$className = _ref.className,
      className = _ref$className === void 0 ? "" : _ref$className,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? "#0d6efd" : _ref$color,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? "3rem" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "3rem" : _ref$height,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? "1s" : _ref$duration,
      others = _objectWithoutPropertiesLoose(_ref, _excluded$a);

  var resolvedWidth = typeof width === "number" ? width + "px" : width;
  var resolvedHeight = typeof height === "number" ? height + "px" : height;
  return React.createElement("svg", Object.assign({}, others, {
    crossOrigin: "anonymous",
    viewBox: "0 0 50 50",
    style: style,
    className: css(_templateObject2$a || (_templateObject2$a = _taggedTemplateLiteralLoose(["\n          width: ", ";\n          height: ", ";\n          animation: ", " ", " ease infinite;\n        "])), resolvedWidth, resolvedHeight, spin$3, duration) + (" " + className)
  }), React.createElement("circle", {
    className: css(_templateObject3$6 || (_templateObject3$6 = _taggedTemplateLiteralLoose(["\n          fill: none;\n          stroke: ", ";\n          opacity: 0.7;\n          stroke-width: 2;\n        "])), color),
    cx: "25",
    cy: "25",
    r: "20"
  }), React.createElement("circle", {
    className: css(_templateObject4$2 || (_templateObject4$2 = _taggedTemplateLiteralLoose(["\n          fill: ", ";\n          stroke: none;\n        "])), color),
    cx: "25",
    cy: "5",
    r: "3.5"
  }));
};

var _excluded$b = ["className", "color", "width", "height", "style", "duration"];

var _templateObject$b, _templateObject2$b;
var spin$4 = /*#__PURE__*/keyframes(_templateObject$b || (_templateObject$b = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n  to {\n    transform: rotate(360deg);\n  }\n"])));

var Spin = function Spin(_ref) {
  var _ref$className = _ref.className,
      className = _ref$className === void 0 ? "" : _ref$className,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? "#0d6efd" : _ref$color,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? "2rem" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "2rem" : _ref$height,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      others = _objectWithoutPropertiesLoose(_ref, _excluded$b);

  var resolvedWidth = typeof width === "number" ? width + "px" : width;
  var resolvedHeight = typeof height === "number" ? height + "px" : height;
  return React.createElement("div", Object.assign({}, others, {
    style: style,
    className: css(_templateObject2$b || (_templateObject2$b = _taggedTemplateLiteralLoose(["\n          border: 3px solid ", "33;\n          border-top-color: ", ";\n          border-radius: 50%;\n          width: ", ";\n          height: ", ";\n          animation: ", " 1s linear infinite;\n        "])), color, color, resolvedWidth, resolvedHeight, spin$4) + (" " + className)
  }));
};

var _excluded$c = ["className", "color", "width", "height", "style", "duration"];

var _templateObject$c, _templateObject2$c;
var spinStretch = /*#__PURE__*/keyframes(_templateObject$c || (_templateObject$c = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n  50% {\n    transform: rotate(360deg) scale(0.4, 0.33);\n    border-width: 8px;\n  }\n  100% {\n    transform: rotate(720deg) scale(1, 1);\n    border-width: 3px;\n  }\n"])));

var SpinStretch = function SpinStretch(_ref) {
  var _ref$className = _ref.className,
      className = _ref$className === void 0 ? "" : _ref$className,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? "#0d6efd" : _ref$color,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? "2rem" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "2rem" : _ref$height,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? "1.2s" : _ref$duration,
      others = _objectWithoutPropertiesLoose(_ref, _excluded$c);

  var resolvedWidth = typeof width === "number" ? width + "px" : width;
  var resolvedHeight = typeof height === "number" ? height + "px" : height;
  return React.createElement("div", Object.assign({}, others, {
    style: style,
    className: css(_templateObject2$c || (_templateObject2$c = _taggedTemplateLiteralLoose(["\n          width: ", ";\n          height: ", ";\n          border: 3px solid transparent;\n          border-top-color: ", ";\n          border-bottom-color: ", ";\n          border-radius: 50%;\n          animation: ", " ", " ease infinite;\n        "])), resolvedWidth, resolvedHeight, color, color, spinStretch, duration) + (" " + className)
  }));
};

var _excluded$d = ["className", "color", "width", "height", "style", "duration"];

var _templateObject$d, _templateObject2$d, _templateObject3$7;
var spin$5 = /*#__PURE__*/keyframes(_templateObject$d || (_templateObject$d = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n  to {\n    transform: rotate(360deg);\n  }\n"])));
var pulse = /*#__PURE__*/keyframes(_templateObject2$d || (_templateObject2$d = /*#__PURE__*/_taggedTemplateLiteralLoose(["\n  from {\n    transform: scale(0.5);\n  }\n  to {\n    transform: scale(1);\n  }\n"])));

var TwinSpin = function TwinSpin(_ref) {
  var _ref$className = _ref.className,
      className = _ref$className === void 0 ? "" : _ref$className,
      _ref$color = _ref.color,
      color = _ref$color === void 0 ? "#0d6efd" : _ref$color,
      _ref$width = _ref.width,
      width = _ref$width === void 0 ? "2rem" : _ref$width,
      _ref$height = _ref.height,
      height = _ref$height === void 0 ? "2rem" : _ref$height,
      _ref$style = _ref.style,
      style = _ref$style === void 0 ? {} : _ref$style,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? "0.6s" : _ref$duration,
      others = _objectWithoutPropertiesLoose(_ref, _excluded$d);

  var resolvedWidth = typeof width === "number" ? width + "px" : width;
  var resolvedHeight = typeof height === "number" ? height + "px" : height;
  return React.createElement("div", Object.assign({}, others, {
    style: style,
    className: css(_templateObject3$7 || (_templateObject3$7 = _taggedTemplateLiteralLoose(["\n          display: flex;\n          width: ", ";\n          height: ", ";\n          border: 3px solid transparent;\n          border-top-color: ", ";\n          border-bottom-color: ", ";\n          border-radius: 50%;\n          animation: ", " calc(", " * 2.5) linear infinite;\n\n          &::before {\n            content: \"\";\n            display: block;\n            margin: auto;\n            width: calc(", " * 0.75 / 3.5);\n            height: calc(", " * 0.75 / 3.5);\n            border: 3px solid ", ";\n            border-radius: 50%;\n            animation: ", " ", " alternate ease-in-out infinite;\n          }\n        "])), resolvedWidth, resolvedHeight, color, color, spin$5, duration, resolvedWidth, resolvedHeight, color, pulse, duration) + (" " + className)
  }));
};

export { BarWave, BouncingBalls, CircularProgress, Coin, FadingBalls, FadingDots, FillingBottle, FlippingSquare, Hypnosis, Messaging, Ring, Spin, SpinStretch, TwinSpin };
